src/alert.cpp

```diff
@@ -1900,7 +1895,7 @@ namespace libtorrent {
 		, m_alloc(alloc)
 		, m_num_peers(int(peers.size()))
 	{
-		std::size_t total_size = m_num_peers; // num bytes for sizes
+		std::size_t total_size = 0; // num bytes for sizes
 		for (int i = 0; i < m_num_peers; i++) {
 			total_size += peers[i].size();
```

Comment:
I'm pretty sure the "underlying type" is the OS type, like ``sockaddr``: http://www.boost.org/doc/libs/1_62_0/doc/html/boost_asio/reference/ip__basic_endpoint/size.html

That's going to me greater than the IP address and port.